Sources/MongoSwiftSync/MongoCursor.swift

```diff
@@ -1,58 +1,126 @@
+     * cursor (e.g. the `maxAwaitTimeMS` option on the `FindOptions` passed to `find`).
+     *
+     * - Returns:
+     *   A `Result<T, Error>?` containing the next `T` in this cursor on success, an error if one occurred, or `nil`
+     *   if there were no results.
+     *
+     *   On failure, there error returned is likely one of the following:
+     *     - `CommandError` if an error occurs while fetching more results from the server.
+     *     - `LogicError` if this function is called after the cursor has died.
+     *     - `LogicError` if this function is called and the session associated with this cursor is inactive.
+     *     - `DecodingError` if an error occurs decoding the server's response.
+     */
+    public func tryNext() -> Result<T, Error>? {
+        do {
+            return try self.asyncCursor.tryNext().wait().map { .success($0) }
+        } catch {
+            return .failure(error)
+        }
     }
 
     /**
-     * Returns the next `Document` in this cursor or `nil`, or throws an error if one occurs -- compared to `next()`,
-     * which returns `nil` and requires manually checking for an error afterward.
-     * - Returns: the next `Document` in this cursor, or `nil` if at the end of the cursor
+     * Returns an array of type `T` from the results of this cursor.
+     *
+     * If this cursor is tailable, this method will block until the cursor is closed or exhausted.
+     *
+     * - Returns: an array of type `T`
      * - Throws:
-     *   - `CommandError` if an error occurs on the server while iterating the cursor.
+     *   - `CommandError` if an error occurs while fetching more results from the server.
      *   - `LogicError` if this function is called after the cursor has died.
      *   - `LogicError` if this function is called and the session associated with this cursor is inactive.
      *   - `DecodingError` if an error occurs decoding the server's response.
      */
-    public func nextOrError() throws -> T? {
-        fatalError("unimplemented")
+    public func all() throws -> [T] {
+        return try self.map {
+            switch $0 {
+            case let .success(t):
+                return t
+            case let .failure(error):
+                throw error
+            }
+        }
     }
 
-    /// Returns the next `Document` in this cursor, or `nil`. After this function returns `nil`, the caller should use
-    /// the `.error` property to check for errors. For tailable cursors, users should also check `isAlive` after this
-    /// method returns `nil`, to determine if the cursor has the potential to return any more data in the future.
-    public func next() -> T? {
-        fatalError("unimplemented")
+    /**
+     * Close this cursor.
+     *
+     * This method may be called from another thread safely even if this cursor is blocked retrieving results.
+     */
+    public func close() {
```

Comment:
We need a public version of this method to allow users to terminate a cursor that's blocked indefinitely.